web/src/users/add_url.rs

```diff
@@ -0,0 +1,41 @@
+use axum::{
+    extract::{Form, State},
+    response::{IntoResponse, Redirect},
+};
+use cja::jobs::Job;
+
+use crate::{jobs::process_article::ProcessArticle, AppState};
+
+use db::{urls::add_url, users::User};
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+pub struct ArticleForm {
+    url: String,
+}
+
+#[axum::debug_handler(state = AppState)]
+pub async fn insert_article_handler(
+    State(state): State<AppState>,
+    user: User,
+    Form(form): Form<ArticleForm>,
+) -> impl IntoResponse {
+    let url = form.url;
+    let user_id = user.user_id;
+
+    match add_url(&state.db, &url, user_id, &true).await {
+        Ok(page) => {
+            let page_id = page.page().page_id;
+            let process_article = ProcessArticle { page_id };
+            process_article
+                .enqueue(state.clone(), "context".to_string())
+                .await
+                .unwrap();
+            Redirect::to(&format!("/article/{page_id}?flash[success]=Article added"))
```

Comment:
Love this!

Just chatting, I think we might want some 'easier' way to make url paths like this in the future. Cause I kinda hate having to do it as string interpolation, but for now this is perfect!
This is mostly a note to myself to think about thinks to build into `cja` more!

Reply:
Alright awesome! Any idea on how we can make this easier?